### PR TITLE
Feature/reset password first page

### PR DESCRIPTION
**Pull request by dmytro-chushko**
Added: 
 - validation and logic to recovery-pass-first-form;
 - recovery-pass-firstHooks;
 - email with token sent and delivered;
Email
Emai_is_required
Email_must_be_valid
Email_does_not_exist
Sending_email

### CODE REVIEW

File: apps/freelance/src/pages/recoverpass-first-page/check-your-email.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { useTranslation } from 'react-i18next';
+import { StyledTitle } from '@freelance/components';
+import { Container } from './recoverpass-first-page.styled';
+
+export function CheckYourEmail() {
+  const { t } = useTranslation();
```
Comment (lyudad) [inline apps/freelance/src/pages/recoverpass-first-page/check-your-email.tsx:6]: Снимок экрана 2022-11-17 в 19 28 08

File: apps/freelance/src/utils/validations/validation-shemas.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { object, string, SchemaOf } from 'yup';
+import { useTranslation } from 'react-i18next';
+
+export interface SendEmailForm {
+  email: string;
+}
+
+export const schemaForgotPasswordForm = (): SchemaOf<SendEmailForm> => {
+  const { t } = useTranslation();
+  const messageEmail: string = t('recoverPassForm.validationEmailMessage');
```
Comment (lyudad) [inline apps/freelance/src/utils/validations/validation-shemas.tsx]: Снимок экрана 2022-11-17 в 19 28 08

File: apps/freelance/src/pages/recoverpass-first-page/check-your-email.tsx (hunk 1 above)
Comment (dmytro-chushko) [reply apps/freelance/src/pages/recoverpass-first-page/check-your-email.tsx:6]: fixed it.

File: apps/freelance/src/utils/validations/validation-shemas.tsx (hunk 2 above)
Comment (dmytro-chushko) [reply apps/freelance/src/utils/validations/validation-shemas.tsx]: fixed it

**Comment by dmytro-chushko** [comment]
add reset password logic and validation.
- recover-pass-sacond-formHook
- resetPasswordScema (validation)
- redux resetPassword mutation
- UI only for testing logic

File: libs/components/src/lib/recover-pass-first-form/recover-pass-firstHooks.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+const CHECK_YOUR_EMAIL_PAGE = '/check-your-email';
+
+export function useForgotPasswordSendEmail(): IonSubmitForgotPassword {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const [sendLinkEmail, { isLoading }] = useSendLinkEmailMutation();
+  const onSubmit: SubmitHandler<IForgotPasswordForm> = async data => {
```
Comment (lyudad) [inline libs/components/src/lib/recover-pass-first-form/recover-pass-firstHooks.ts:16]: Снимок экрана 2022-11-20 в 19 58 52

File: libs/components/src/lib/recover-pass-second-form/recover-pass-second-form.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  const {
+    register,
+    setFocus,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<IResetPasswordForm>({ resolver: yupResolver(schema) });
+  const { onSubmit, isLoading } = useResetPassword();
+  useEffect(() => {
+    setFocus('password');
+  }, [setFocus]);
```
Comment (lyudad) [inline libs/components/src/lib/recover-pass-second-form/recover-pass-second-form.tsx]: Снимок экрана 2022-11-20 в 19 58 52

File: libs/components/src/lib/recover-pass-first-form/recover-pass-firstHooks.ts (hunk 3 above)
Comment (dmytro-chushko) [reply libs/components/src/lib/recover-pass-first-form/recover-pass-firstHooks.ts:16]: fixed

File: libs/components/src/lib/recover-pass-second-form/recover-pass-second-form.tsx (hunk 4 above)
Comment (dmytro-chushko) [reply libs/components/src/lib/recover-pass-second-form/recover-pass-second-form.tsx]: fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**